xchange-bitmex/src/main/java/org/knowm/xchange/bitmex/service/BitmexAccountServiceRaw.java

```diff
@@ -0,0 +1,81 @@
+package org.knowm.xchange.bitmex.service;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.bitmex.BitmexException;
+import org.knowm.xchange.bitmex.dto.account.BitmexAccount;
+import org.knowm.xchange.bitmex.dto.account.BitmexMarginAccount;
+import org.knowm.xchange.bitmex.dto.account.BitmexWallet;
+import org.knowm.xchange.bitmex.dto.account.BitmexWalletTransaction;
+import org.knowm.xchange.currency.Currency;
+import org.knowm.xchange.exceptions.ExchangeException;
+
+public class BitmexAccountServiceRaw extends BitmexBaseService {
+
+  /**
+   * Constructor
+   *
+   * @param exchange
+   */
+  public BitmexAccountServiceRaw(Exchange exchange) {
+
+    super(exchange);
+  }
+
+  public BitmexAccount getBitmexAccountInfo() throws IOException {
+
+    try {
+      return bitmex.getAccount(apiKey, exchange.getNonceFactory(), signatureCreator);
+    } catch (BitmexException e) {
+      throw new ExchangeException(e);
+    }
+  }
+
+  public BitmexWallet getBitmexWallet(Currency ccy) throws IOException {
+
+    try {
+      return bitmex.getWallet(apiKey, exchange.getNonceFactory(), signatureCreator, ccy.getCurrencyCode());
+    } catch (BitmexException e) {
+      throw new ExchangeException(e);
+    }
+  }
+
+  public List<BitmexWalletTransaction> getBitmexWalletHistory(Currency ccy) throws IOException {
+
+    try {
+      return bitmex.getWalletHistory(apiKey, exchange.getNonceFactory(), signatureCreator, ccy.getCurrencyCode());
+    } catch (BitmexException e) {
+      throw new ExchangeException(e);
+    }
+  }
+
+  public List<BitmexWalletTransaction> getBitmexWalletSummary(Currency ccy) throws IOException {
+
+    try {
+      return bitmex.getWalletSummary(apiKey, exchange.getNonceFactory(), signatureCreator, ccy.getCurrencyCode());
+    } catch (BitmexException e) {
+      throw new ExchangeException(e);
+    }
+  }
+
+  public BitmexMarginAccount getBitmexMarginAccountStatus(Currency ccy) throws IOException {
+
+    try {
+      return bitmex.getMarginAccountStatus(apiKey, exchange.getNonceFactory(), signatureCreator, ccy.getCurrencyCode());
+    } catch (BitmexException e) {
+      throw new ExchangeException(e);
+    }
+  }
+
+  public List<BitmexMarginAccount> getBitmexMarginAccountsStatus() throws IOException {
+
+    try {
+      return bitmex.getMarginAccountsStatus(apiKey, exchange.getNonceFactory(), signatureCreator);
+    } catch (BitmexException e) {
+      throw new ExchangeException(e);
```

Comment:
You should handle the exceptions with a handle exceptions method in BitmexBaseService. See GDAX impl. on develop. That way, it's easier for clients to handle certain types of exceptions.

Reply:
done